core/domain/app_feedback_report_validators_test.py

```diff
@@ -0,0 +1,728 @@
+            'entry_point_name': 'crash',
+        },
+        'text_size': 'MEDIUM_TEXT_SIZE',
+        'download_and_update_only_on_wifi': True,
+        'automatically_update_topics': False,
+        'is_admin': False
+    }
+    ANDROID_REPORT_INFO_SCHEMA_VERSION = 1
+
+    def setUp(self):
+        super(AppFeedbackReportModelValidatorTests, self).setUp()
+        self.signup(self.USER_1_EMAIL, self.USER_1_USERNAME)
+        self.user_1_id = self.get_user_id_from_email(self.USER_1_EMAIL)
+        self.report_id = (
+            app_feedback_report_models.AppFeedbackReportModel.create(
+                platform=self.PLATFORM_ANDROID,
+                submitted_on=self.REPORT_SUBMITTED_TIMESTAMP,
+                report_type=self.REPORT_TYPE_SUGGESTION,
+                category=self.CATEGORY_OTHER,
+                platform_version=self.PLATFORM_VERSION,
+                android_device_country_locale_code=(
+                    self.COUNTRY_LOCALE_CODE_INDIA),
+                android_sdk_version=self.ANDROID_SDK_VERSION,
+                android_device_model=self.ANDROID_DEVICE_MODEL,
+                entry_point=self.ENTRY_POINT_NAVIGATION_DRAWER,
+                entry_point_topic_id=None,
+                entry_point_story_id=None,
+                entry_point_exploration_id=None,
+                entry_point_subtopic_id=None,
+                text_language_code=self.TEXT_LANGUAGE_CODE_ENGLISH,
+                audio_language_code=self.AUDIO_LANGUAGE_CODE_ENGLISH,
+                android_report_info=self.ANDROID_REPORT_INFO,
+                web_report_info=None))
+
+        self.expected_output = [
+            u'[u\'fully-validated AppFeedbackReportModel\', 1]']
+        self.job_class = (
+            prod_validation_jobs.AppFeedbackReportModelAuditOneOffJob)
+
+    def test_standard_model(self):
+        self.run_job_and_check_output(
+            self.expected_output, sort=False, literal_eval=False)
+
+    def test_model_validation_with_valid_external_references_succesful(self):
+        model_entity = (
+            app_feedback_report_models.AppFeedbackReportModel.get_by_id(
+                self.report_id))
+        model_entity.ticket_id = self.TICKET_ID
+        model_entity.update_timestamps()
+        model_entity.put()
+        app_feedback_report_models.AppFeedbackReportTicketModel(
+            id=self.TICKET_ID, ticket_name='example ticket',
+            github_issue_repo_name=None, github_issue_number=None,
+            archived=False,
+            newest_report_timestamp=self.REPORT_SUBMITTED_TIMESTAMP,
+            report_ids=[self.report_id]).put()
+
+        self.run_job_and_check_output(
+            self.expected_output, sort=False, literal_eval=False)
+
+    def test_model_validation_with_invalid_external_references_fails(self):
+        model_entity = (
+            app_feedback_report_models.AppFeedbackReportModel.get_by_id(
+                self.report_id))
+        model_entity.ticket_id = 'invalid_ticket_id'
```

Comment:
Here & below: suggest also adding tests for deleted reports for invalid (vs. just testing nonexistent). Those cases seem missing from the behavioral coverage.

Reply:
Do you mean that models will be hard-deleted after a certain period of time, is this applied to all models? We don't delete reports anywhere.

Reply:
Done